Expose Methods so other Plugins can hook

DrkMatr1984 opened the pull request:
into dsTitles to retrieve data for Chat PlaceHolders(ie. DeluxeChat),
add getChatTag method, bump to 1.5 in Plugin.yml(Shouldv'e done this
already)

DrkMatr1984 committed:
Expose Methods so other Plugins can hook

into dsTitles to retrieve data for Chat PlaceHolders(ie. DeluxeChat),
add getChatTag method, bump to 1.5 in Plugin.yml(Shouldv'e done this
already)

DrkMatr1984 commented:
Still have to go through and change getPlayer(name) references to getPlayer(player) or getPlayer(UUID) to provide full UUID compatibility.

DrkMatr1984 committed:
Fix Possible NPE that occurs when versionChecker = null and an admin logs in""

and an admin logs in. Also added some @SuppressWarnings("deprecation")
Suppressions

DrkMatr1984 committed:
Fix Possible NPEs

that sometimes occured when a player logged in without a title or tried
to chat without a title. Now, everyone will be assigned the default
prefix and suffix if they have been defined in the config.

Fixed NPE with getChatTag() method.

DrkMatr1984 committed:
Clean up Code, add DeluxeChat/PlaceholderAPI Hooks/chat

placeholders

DrkMatr1984 committed:
Update pom.xml

denniss17 commented:
Wow I never noticed this PR untill now, so sorry for not responding :P. Is this PR still relevant?

DrkMatr1984 commented:
Yup, I'm running it currently on my 1.9 server. I did see the issue that other guy posted... I guess they changed the getOnlinePlayers() method in 1.9. I didn't get around to updating that yet, but it seems to function correctly on my server. I'd take the PR and merge and try to see if you can fix that, or I'll get around to it sooner or later.

DrkMatr1984 commented:
I just realized the way I have it, you'll need DeluxeChat, a spigot premium resource, to compile it. As long as you don't redistribute it, I can give you a copy for development purposes. Grab it from 
https://www.dropbox.com/s/baddtuzg1gf7qnd/Dependencies.zip?dl=0

DrkMatr1984 committed:
Fix Loading Issues when DeluxeChat or PlaceholderAPI were not installed

DrkMatr1984 committed:
Move Placeholders into sub-package

Getting ready to also add mvdw's placeholderAPI support. Moved
placeholders to their own sub-package.

DrkMatr1984 committed:
Update ReadMe

DrkMatr1984 committed:
Update Readme

DrkMatr1984 committed:
Update ReadMe

DrkMatr1984 committed:
Fix invis players showing nametags

All players are now put into their own team to fix issues with nametags
being visible to other players, even if player is invis.

DrkMatr1984 committed:
Add support for mvdwPlaceHolders

Also fixed once and for all the problem of not having both DeluxeChat
and PlaceHolderAPI installed. mvdwPlaceHolderAPI support is untested.
Bumped version to 1.5.2

DrkMatr1984 committed:
Oops, forgot to remove this from Git directory

DrkMatr1984 committed:
Finalized mvdwPlaceholderAPI support

Updated Readme

DrkMatr1984 committed:
Remove Debug Messages

DrkMatr1984 commented:
> Yup, I'm running it currently on my 1.9 server. I did see the issue that other guy posted... I guess they changed the getOnlinePlayers() method in 1.9. I didn't get around to updating that yet, but it seems to function correctly on my server. I'd take the PR and merge and try to see if you can fix that, or I'll get around to it sooner or later.

Yep, this is fixed. As well as a few other bugs. I believe it would even be backwards compatible to pretty far back.

DrkMatr1984 committed:
Update Repo to Current Source

Update repo to current local source code

DrkMatr1984 committed:
Update README.md

DrkMatr1984 committed:
Fix small Error

DrkMatr1984 committed:
Fix SQL

Fix SQL, forgot to bump version is Plugin.yml

DrkMatr1984 committed:
Update titleConfig.yml

Add some better examples

DrkMatr1984 committed:
Fix Clicking on Suffix errors out

DrkMatr1984 committed:
1.17.1 Update

Update to 1.17.1 Sources, remove Deluxechat hook and softdepend.

DrkMatr1984 committed:
Fix it

Reworked Configs, changed over to UUIDs for storing player data, fixed bugs

DrkMatr1984 committed:
Create PluginDescriptionSpigot.odt

Create a file to manage the Spigot Page

DrkMatr1984 committed:
Reworked Permissions, Console access to Commands

Additional changes to config, commands reworked a little bit.

DrkMatr1984 committed:
Fix NPE

DrkMatr1984 commented:
A lot more fixes aded now, if you still want to merge the pull request.

DrkMatr1984 committed:
Added Prefix/Suffix "Symbols", Code Tidy-up

I've added prefix and suffix "symbols", basically a one letter abbreviation of your Suffix or Prefix. (I use mine with Wolfie Mario's Font Pack to give each title an Icon, personally)

DrkMatr1984 committed:
Code Cleanup

DrkMatr1984 committed:
Update README.md

DrkMatr1984 committed:
Update README.md

DrkMatr1984 committed:
Update README.md

DrkMatr1984 committed:
Save Data Async

DrkMatr1984 committed:
Support Hex Colors

Hex Colors are now supported in chatTag and description.
Use them like <#2AA8F2> and
<gradient:#2AA8F2:#8BD448>

DrkMatr1984 committed:
Update ReadME and License

DrkMatr1984 committed:
Added HikariCP connection pooling

DrkMatr1984 committed:
Simplify Code, Fix lock time out mysql errors

DrkMatr1984 commented:
To the Moon! 🚀

DrkMatr1984 committed:
Create LICENSE_2

HikariCP License

DrkMatr1984 committed:
Include Licenses in Jar

DrkMatr1984 committed:
Bump Version, Fix SQLite

DrkMatr1984 committed:
Fix shutdown error, Handle SQLite properly

DrkMatr1984 committed:
Code Improvements, Fix MySQL timing out
